mediator/med_phases_prep_atm_mod.F90

```diff
@@ -245,7 +245,7 @@ subroutine med_phases_prep_atm(gcomp, rc)
     end if
 
     ! Check for nans in fields export to atm
-    call FB_check_for_nans(is_local%wrap%FBExp(compatm), rc=rc)
+    call FB_check_for_nans(gcomp, is_local%wrap%FBExp(compatm), maintask, logunit, rc=rc)
```

Comment:
I don't understand why you are passing gcomp here. It seems that you need an initialization call to determine checkfornans as a module variable in med_methods_mod and then you don't need to pass any gcomp to the subroutines.
At least that is what I am subbesting.